src/cryptsetup/cryptsetup-tokens/cryptsetup-token-systemd-fido2.c

```diff
@@ -49,7 +49,7 @@ _public_ int cryptsetup_token_open_pin(
         /* pin was passed as pin = pin, pin_size = strlen(pin). We need to add terminating
          * NULL byte to addressable memory*/
         if (pin && pin[pin_size-1] != '\0') {
-                pin_string = strndup(pin, pin_size);
+                pin_string = memdup_suffix0(pin, pin_size);
                 if (!pin_string)
                         return crypt_log_oom(cd);
         }
```

Comment:
actually, given this same code is used multiple times we should make this a common helper?

```c
static int normalize_pin(const void *pin, size_t pin_size, char **ret) {
…
}
```
That first checks for embedded nUl, and then suffixes the NUL byte? and returns the resulting string

Reply:
Sounds good, implemented in https://github.com/systemd/systemd/blob/559a1d35d510e3ae637953b1ae9bfab4a124c2f8/src/cryptsetup/cryptsetup-tokens/cryptsetup-token-util.c#L60-L84